python-sdk/src/astro/sql/operators/data_validations/ColumnCheckOperator.py

```diff
@@ -0,0 +1,242 @@
+
+            # Iterating over checks
+            for check in checks:
+                tolerance = self.column_mapping[column][check].get("tolerance")
+                result = self.get_check_result(check, column_name=column, df=self.df)
+                self.column_mapping[column][check]["result"] = result
+                self.column_mapping[column][check]["success"] = self._get_match(
+                    self.column_mapping[column][check], result, tolerance
+                )
+                failed_tests.extend(_get_failed_checks(self.column_mapping[column], column))
+                passed_tests.extend(_get_success_checks(self.column_mapping[column], column))
+
+        if len(failed_tests) > 0:
+            raise AirflowException(f"The following tests have failed:" f"\n{''.join(failed_tests)}")
+        if len(passed_tests) > 0:
+            print(f"The following tests have passed:" f"\n{''.join(passed_tests)}")
+
+    @staticmethod
+    def col_null_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
+        """
+        Count the total null values in a dataframe column
+        """
+        return list(df[column_name].isnull().values).count(True)
+
+    @staticmethod
+    def col_distinct_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
+        """
+        Count the distinct value in a dataframe column
+        """
+        return len(df[column_name].unique())
+
+    @staticmethod
+    def col_unique_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
+        """
+        Count the unique value in a dataframe column
+        """
+        return len(df[column_name]) - len(df[column_name].unique())
+
+    @staticmethod
+    def col_max(column_name: str, df: pandas.DataFrame) -> Optional[float]:
+        """
+        Get the max value in dataframe column
+        """
+        return df[column_name].max()
+
+    @staticmethod
+    def col_min(column_name: str, df: pandas.DataFrame) -> Optional[float]:
+        """
+        Get the min value in dataframe column
+        """
+        return df[column_name].min()
+
+
+def _get_failed_checks(checks, col=None):
+    if col:
+        return [
+            f"Column: {col}\nCheck: {check},\nCheck Values: {check_values}\n"
+            for check, check_values in checks.items()
+            if not check_values["success"]
+        ]
+    return [
+        f"\tCheck: {check},\n\tCheck Values: {check_values}\n"
+        for check, check_values in checks.items()
+        if not check_values["success"]
+    ]
+
+
+def _get_success_checks(checks, col=None):
+    if col:
+        return [
+            f"Column: {col}\nCheck: {check},\nCheck Values: {check_values}\n"
+            for check, check_values in checks.items()
+            if check_values["success"]
+        ]
+    return [
+        f"\tCheck: {check},\n\tCheck Values: {check_values}\n"
+        for check, check_values in checks.items()
+        if check_values["success"]
+    ]
```

Comment:
We have a lot of duplicated code here, maybe we can create a function to define the `f"Column: {col}\nCheck: {check},\nCheck Values: {check_values}\n"` which retrieves the needed args. The same for the condition. WDYT?